qiskit_experiments/characterization/t2star_experiment.py

```diff
@@ -82,20 +82,22 @@ def _format_plot(ax, unit):
             ax.set_ylabel("Probability to measure |0>", fontsize=12)
 
         # implementation of  _run_analysis
-        unit = experiment_data._data[0]["metadata"]["unit"]
-        conversion_factor = experiment_data._data[0]["metadata"].get("dt_factor", None)
+        data = experiment_data.data()
+        unit = data[0]["metadata"]["unit"]
+        conversion_factor = data[0]["metadata"].get("dt_factor", None)
         if conversion_factor is None:
-            conversion_factor = 1 if unit == "s" else apply_prefix(1, unit)
+            conversion_factor = 1 if unit in ("s", "dt") else apply_prefix(1, unit)
+
         xdata, ydata, sigma = process_curve_data(
             experiment_data._data, lambda datum: level2_probability(datum, "0")
         )
 
-        si_xdata = xdata * conversion_factor
+        si_xdata = xdata
         t2star_estimate = np.mean(si_xdata)
-
         p0, bounds = self._t2star_default_params(
             conversion_factor, user_p0, user_bounds, t2star_estimate
         )
+        si_xdata *= conversion_factor
```

Comment:
done

Reply:
The frequency cannot be 0 because I define the bounds as multiples of the frequency, and if it is 0, then the bounds will be [0, 0] which is not acceptable.